watson/utils.py

```diff
@@ -111,3 +114,44 @@ def get_frame_from_argument(watson, arg):
             style('error', "No frame found with id"),
             style('short_id', arg))
         )
+
+
+def make_json_writer(func, *args, **kwargs):
```

Comment:
I wanted to handle the two cases of `config.write(fp)` and `json.dump(data, fp)` with a common interface, so `safe_save` expects a function, which takes a file-object and knows how to write some data to it. This can be `config.write`, for example, or the function returned by `make_json_writer`.

I wanted to avoid having to store the data, which the function writes, in memory, so instead of passing the data to `make_json_writer`, you pass a function, which generates the data to write, for example `frames.dump`. The effect is that the function, which generates the data, is only called at the moment the writer function passed to `safe_save` is called. If the data generation function is a e.g. a generator and the writer function can take advantage of it, this makes writing more efficient than passing the whole lump of data as a variable.